data_warehouse/views.py

```diff
@@ -28,3 +37,33 @@ def get_canvas_options(self, request):
         '''getting canvas-options'''
         data = CanvasOptionsService.get_canvas_options()
         return Response(data)
+
+    @action(detail=False, methods=['GET'], url_path='table-export',
+            url_name='table-export')
+    def export(self, request):
+        '''Return a xls file stream.'''
+        if 'id' not in request.GET:
+            raise BadRequest('请求的参数不正确。')
+        method_name = request.GET.get('method_name')
```

Comment:
If you take the suggestions, then I think you don't need the `method_name`, cuz you can determine which function should be called based on `table_type` in your dispatch function as suggested above.